strapi/strapi
# Update "upgrade major" rules

### What does it do?

Refines the logic for calculating version ranges for `major` version upgrades:
- Update the major range from `<= next_major.0.0` to `<= next_major` , which translates to `< next_next_major.0.0-0`
- Include more intuitive comments to clarify the purpose and behavior of each range calculation.

### Why is it needed?

Targets the latest `minor.patch` when doing a `major` upgrade instead of targeting the first `patch` of the next `major`.
This helps mitigate migration issues from early versions of new majors.

### How to test it?

1. Run the updated unit tests in `range.test.ts`.
2. Verify that the new ranges (`<=2` instead of `<=2.0.0`) are generated correctly in different scenarios.

**or**

Test migrating a `v4` application on latest `minor` to the latest `v5` patch (`5.0.4` at the time of writing this)
### Related issue(s)/PR(s)
